File: sample/src/main/java/com/sys1yagi/swipe/sample/tools/AssetsPathStore.java
package com.sys1yagi.swipe.sample.tools;

import com.sys1yagi.swipe.sample.util.AssetsUtils;

import android.content.res.AssetManager;

import java.util.ArrayList;
import java.util.List;

public class AssetsPathStore {

    AssetsUtils assetsUtils = new AssetsUtils();

    private List<String> pathList = new ArrayList<>();

    public AssetsPathStore(AssetManager assetManager) {
        assetsUtils.scan(assetManager, pathList);
    }

    public boolean contains(String name) {
        for (String path : pathList) {
            if (path.lastIndexOf(name) != -1) {
                return true;
            }
        }
        return false;
    }

    public String toPath(String name) {
        for (String path : pathList) {
            if (path.lastIndexOf(name) != -1) {
                return path;
            }
        }
        return null;
    }

    public String getAssetPath(String name) {
        if (!contains(name)) {
            return null;
        }
        return "file:///android_asset/" + toPath(name);
    }

}

File: core/src/main/java/com/sys1yagi/swipe/core/tool/AssetsPathStore.kt
package com.sys1yagi.swipe.core.tool


import android.content.res.AssetManager
import android.text.TextUtils
import com.sys1yagi.swipe.core.util.AssetsUtils
import java.util.*

class AssetsPathStore(assetManager: AssetManager) {

    internal var assetsUtils = AssetsUtils()

    private val pathList = ArrayList<String>()

    init {
        assetsUtils.scan(assetManager, pathList)
    }

    operator fun contains(name: String?): Boolean {
        return !TextUtils.isEmpty(toPath(name))
    }

    fun toPath(name: String?): String? {
        name?.let {
            for (path in pathList) {
                if (path.lastIndexOf(it) != -1) {
                    return path
                }
            }
        }
        return null
    }

    fun getAssetPath(name: String?): String? {
        if (!contains(name)) {
            return null
        }
        return "file:///android_asset/" + toPath(name)!!
    }

}
